contracts/airdrop/CruzoAirdrop.sol

```diff
@@ -0,0 +1,100 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/IERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/utils/ERC1155HolderUpgradeable.sol";
+
+import "../transfer-proxy/ITransferProxy.sol";
+
+contract CruzoAirdrop is
+    Initializable,
+    UUPSUpgradeable,
+    OwnableUpgradeable,
+    ERC1155HolderUpgradeable
+{
+    using CountersUpgradeable for CountersUpgradeable.Counter;
+
+    struct Drop {
+        address tokenAddress;
+        uint256 tokenId;
+        uint256 amount;
+        uint256 claimed;
+        mapping(address => bool) claimers;
+    }
+
+    event DropCreated(
+        uint256 id,
+        address tokenAddress,
+        uint256 tokenId,
+        address creator,
+        uint256 amount
+    );
+
+    event DropClaimed(uint256 id, address claimer);
+
+    CountersUpgradeable.Counter private ids;
+
+    ITransferProxy public transferProxy;
+
+    mapping(uint256 => Drop) public drops;
+
+    constructor() {}
+
+    function _authorizeUpgrade(address) internal override onlyOwner {}
+
+    function initialize(ITransferProxy _transferProxy) public initializer {
+        __Ownable_init();
+        transferProxy = _transferProxy;
+    }
+
+    function claim(uint256 _id) external {
+        Drop storage drop = drops[_id];
+        require(drop.amount > 0, "Airdrop: not found");
+        require(drop.claimed < drop.amount, "Airdrop: closed");
+        require(!drop.claimers[_msgSender()], "Airdrop: already claimed");
```

Comment:
In this function msg.sender func is called 3 times, isn't it cheaper to get this value at the beginning of the function?